Pear/Settings/EditLocationAvailabilityViewController.swift

```diff
@@ -206,18 +223,28 @@ class EditLocationAvailabilityViewController: UIViewController {
 extension EditLocationAvailabilityViewController: UICollectionViewDataSource {
 
     func numberOfSections(in collectionView: UICollectionView) -> Int {
-        2
+        3
     }
 
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         switch locationSections[section] {
+        case .online(let locations): return locations.count
         case .campus(let locations): return locations.count
         case .ctown(let locations): return locations.count
         }
     }
```

Comment:
I've seen this switch statement more than once.  It seems very redundant to have 3 cases that do the exact same thing.  I would either create a function that does this or try and assign locations as a variable within the enum so the redundancy only occurs once.  Below is an example.

```
enum LocationSection {
    case online([String])
    case campus([String])
    case ctown([String])

    var locations: [String] {
        switch self {
        case .online(let locationArr): return locationArr
        case .campus(let locationArr): return locationArr
        case .ctown(let locationArr): return locationArr
        }
    }
}

func example(numberOfItemsInSection section: Int) -> Int {
     return locationSections[section].locations.count
}

```
Doing this would reduce redundancy here and in SchedulingPlacesViewController lines 372-377.  This would also keep you from writing (let locations) every time you have a switch statement with Section

I'm not sure if there's a better way to do what I'm suggesting...maybe there is I'm not sure.  This is the best way I could think of though.

Reply:
This class is _very_ similar to SchedulingPlacesViewController. Can you refactor and extract shared components?